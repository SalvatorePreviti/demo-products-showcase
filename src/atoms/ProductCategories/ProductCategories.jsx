import React from 'react'
import PropTypes from 'prop-types'
import styles from './ProductCategories.scss'

const itemStyle = styles.item
const itemActiveStyle = styles.item + ' ' + styles.active

function ProductCategories(props) {
  const { categories, activeCategory, onSelectCategory } = props
  return (
    <header>
      Store Cupboard
      {categories.map(category => {
        return (
          <a
            key={category.id}
            href="#"
            className={category === activeCategory ? itemActiveStyle : itemStyle}
            onClick={() => onSelectCategory && onSelectCategory(category)}>
            {category.title}
          </a>
        )
      })}
    </header>
  )
}

const categoryType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
})

ProductCategories.propTypes = {
  categories: PropTypes.arrayOf(categoryType.isRequired).isRequired,
  activeCategory: categoryType,
  onSelectCategory: PropTypes.func
}

export default ProductCategories
