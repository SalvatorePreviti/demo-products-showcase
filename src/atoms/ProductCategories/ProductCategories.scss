.item {
  display: inline-block;
  margin-left: 10px;
  padding: 5px;
  text-decoration: none;

  &,
  &:visited,
  &:active,
  &:focus {
    color: #000;
  }
}

.active {
  font-weight: bold;
  text-decoration: underline;
}
