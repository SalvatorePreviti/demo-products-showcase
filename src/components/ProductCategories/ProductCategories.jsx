import React from 'react'
import PropTypes from 'prop-types'
import styles from './ProductCategories.scss'

const itemStyle = styles.item
const itemActiveStyle = styles.item + ' ' + styles.active

function ProductCategories(props) {
  const { categories, activeCategoryId, onSelectCategory } = props
  return (
    <header>
      Store Cupboard
      {categories.map(category => {
        const categoryId = category.id
        return (
          <a
            key={categoryId}
            href="#"
            className={categoryId === activeCategoryId ? itemActiveStyle : itemStyle}
            onClick={() => onSelectCategory && onSelectCategory(categoryId)}>
            {category.title}
          </a>
        )
      })}
    </header>
  )
}

ProductCategories.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired
    })
  ).isRequired,
  activeCategoryId: PropTypes.string,
  onSelectCategory: PropTypes.func
}

export default ProductCategories
