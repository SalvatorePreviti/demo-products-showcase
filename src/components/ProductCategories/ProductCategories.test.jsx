import React from 'react'
import Renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'

import ProductCategories from './ProductCategories'

describe('ProductCategories', () => {
  const categories = [
    { id: '1', title: 'title 1' },
    { id: '2', title: 'title 2' },
    { id: '3', title: 'title 3' },
    { id: '4', title: 'title 4' }
  ]

  describe('with a list of categories, without selection', () => {
    function getComponent() {
      return <ProductCategories categories={categories} />
    }

    it('snapshots', () => {
      const shallowRenderer = new ShallowRenderer()
      shallowRenderer.render(getComponent())
      expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
    })

    it('renders all elements', () => {
      const root = Renderer.create(getComponent()).root
      const found = root.findAll(item => item.type === 'a', { deep: true })

      expect(found.map(item => item.props)).toEqual([
        { href: '#', className: 'item', children: 'title 1' },
        { href: '#', className: 'item', children: 'title 2' },
        { href: '#', className: 'item', children: 'title 3' },
        { href: '#', className: 'item', children: 'title 4' }
      ])
    })
  })

  describe('with a list of categories, with a selection', () => {
    function getComponent() {
      return <ProductCategories categories={categories} activeCategoryId="2" />
    }

    it('snapshots', () => {
      const shallowRenderer = new ShallowRenderer()
      shallowRenderer.render(getComponent())
      expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
    })

    it('renders all elements, marking as active the active element', () => {
      const root = Renderer.create(getComponent()).root
      const found = root.findAll(item => item.type === 'a', { deep: true })
      expect(found.map(item => item.props)).toEqual([
        { href: '#', className: 'item', children: 'title 1' },
        { href: '#', className: 'item active', children: 'title 2' },
        { href: '#', className: 'item', children: 'title 3' },
        { href: '#', className: 'item', children: 'title 4' }
      ])
    })
  })
})
