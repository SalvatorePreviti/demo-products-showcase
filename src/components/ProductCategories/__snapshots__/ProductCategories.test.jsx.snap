// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategories with a list of categories, with a selection snapshots 1`] = `
<header>
  Store Cupboard
  <a
    className="item"
    href="#"
    key="1"
    onClick={[Function]}
  >
    title 1
  </a>
  <a
    className="item active"
    href="#"
    key="2"
    onClick={[Function]}
  >
    title 2
  </a>
  <a
    className="item"
    href="#"
    key="3"
    onClick={[Function]}
  >
    title 3
  </a>
  <a
    className="item"
    href="#"
    key="4"
    onClick={[Function]}
  >
    title 4
  </a>
</header>
`;

exports[`ProductCategories with a list of categories, without selection snapshots 1`] = `
<header>
  Store Cupboard
  <a
    className="item"
    href="#"
    key="1"
    onClick={[Function]}
  >
    title 1
  </a>
  <a
    className="item"
    href="#"
    key="2"
    onClick={[Function]}
  >
    title 2
  </a>
  <a
    className="item"
    href="#"
    key="3"
    onClick={[Function]}
  >
    title 3
  </a>
  <a
    className="item"
    href="#"
    key="4"
    onClick={[Function]}
  >
    title 4
  </a>
</header>
`;
